src/ORM/Behavior/CounterCacheBehavior.php

```diff
@@ -114,7 +115,7 @@ class CounterCacheBehavior extends Behavior
      * @param \ArrayObject $options The options for the query
      * @return void
      */
-    public function beforeSave(Event $event, EntityInterface $entity, $options)
+    public function beforeSave(Event $event, EntityInterface $entity, ArrayObject $options)
```

Comment:
Can we hint against the `ArrayAccess` interface instead of a concrete implementation?